deployer/deployment_deployer.go

```diff
@@ -74,20 +74,15 @@ func (d *DeploymentDeployer) Deploy(ctx context.Context, dl *workloads.Deploymen
 		return errors.Wrap(err, "couldn't generate deployments data")
 	}
 
-	oldDeployments := d.tfPluginClient.StateLoader.currentNodeDeployment
-
-	currentDeployments, err := d.deployer.Deploy(ctx, oldDeployments, newDeployments, newDeploymentsSolutionProvider)
+	dl.NodeDeploymentID, err = d.deployer.Deploy(ctx, dl.NodeDeploymentID, newDeployments, newDeploymentsSolutionProvider)
```

Comment:
We should never do that before checking on the error, `dl` is a pointer, meaning the change here will happen regardless of the error check, and the caller will end up in an inconsistent state. Let's change the passed in arguments if and only if error is nil.